src/EnumDefinition.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Spatie\Enum;
+
+class EnumDefinition
+{
+    /** @var string|int */
+    public $value;
+
+    public string $label;
+
+    private string $methodName;
```

Comment:
By having this property `private` this class will be unextendable because the key information isn't accessible for extending classes. So I would be fine with making the whole class also `final` - indicating more prominent that there's no good way to extend it and that it's only an internal value object.

Reply:
I totally agree, but @freekmurze doesn't, which means I'm stuck 😭

Reply:
I know that @freekmurze doesn't like `final` which is okay for me and I also use it very very very limited in packages. But if everything is private the class is like `final` because even if you extend you start with a blank canvas - unable to call/retrieve anything from the parent.

So I'm okay with make everything `protected` to keep it extendable - for whatever reason. Or make it `final`, which I prefer, and also add `@internal` annotation to indicate that this class is like invisible and should never be used by the user.